GSA/openacr-editor
# WCAG 2.1 support and switching between catalogs

Closes https://github.com/GSA/openacr/issues/270

Summary of changes:
* Moved catalog loading to a separate file+function. Added new catalogs from @openacr/openacr.
* Added a catalog radio field to the overview page to allow switching of catalogs.
* Added code to handle when switching catalogs to add missing fields and remove fields (e.g. remove WCAG 2.1 fields when switching to WCAG 2.0 to pass validation).
* Added tests.

**QA new report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. On the overview page confirm you see 'Select catalog' accordion.
4. Confirm the accordion is expanded and has a radio choices below the text 'Select which catalog you will be using for the OpenACR.'
5. Confirm the choices are 'VPAT® 2.4 Revised Section 508 Edition (WCAG 2.0)', 'VPAT® 2.4 WCAG 2.1 and Revised Section 508 Edition', 'VPAT® 2.4 (WCAG 2.1)'.
6. Switch the choice from WCAG 2.0 to any of the other choices.
7. Confirm the chapters in the navigation changes and the sidebar changes when switching the choices.
8. Selecting WCAG 2.1 catalog.
9. Navigate to the chapter pages and confirm you see WCAG 2.1 criteria and links go to WCAG 2.1 pages.
10. Add some text, select some values.
11. Click the 'View report' or 'Report' tab.
12. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
13. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
14. Download the YAML & HTML report.
15. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
16. Go back to the 'Overview' page
17. Click 'Start new report' or 'New report'.
18. Click 'Open report'.
19. Upload the downloaded YAML file from step 14.
20. Confirm the catalog selection shows the WCAG 2.1 selection.

**QA existing report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Click 'Open report'.
4. Upload a yaml file from OpenACR that already has 'last modified date' and 'version' set to some values (if needed modify the YAML file to set those values before uploading). Try the 'drupal-9.yaml' or any of the yaml files from https://github.com/GSA/openacr/tree/main/openacr where last modified date is 11/16/2021 and version is 12.
5. Confirm the catalog selection shows the WCAG 2.0 selection.
6. Switch the choice from WCAG 2.0 to any of the other choices.
7. Navigate to the chapter pages and confirm you see WCAG 2.1 criteria and links go to WCAG 2.1 pages.
8. Add some text, select some values.
9. Click the 'View report' or 'Report' tab.
10. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
11. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
12. Download the YAML & HTML report.
13. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
14. Go back to the 'Overview' page
15. Click 'Start new report' or 'New report'.
16. Upload the downloaded YAML file from step 12.
17. Confirm the catalog selection shows the WCAG 2.1 selection.
18. Switch the choice **back to WCAG 2.0**.
19. Navigate to the chapter pages and confirm you **only** see WCAG 2.0 criteria and links go to WCAG 2.0 pages.
20. Click the 'View report' or 'Report' tab.
21. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
22. Confirm the report has the **only** see WCAG 2.0 criteria and links go to WCAG 2.0 pages.
